### PR TITLE
Add sailing-weather intent & testing

### PR DESCRIPTION
**Pull request by GreenGiraffe1**
This intent provides users the sailing weather at
Community Boating (CBI) on the Charles River.
Weather information is queried from CBI's API.

Addresses Issue #156

### CODE REVIEW

File: mycity/mycity/intents/sailing_weather_intent.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        'MyCityRequestDataModel received:',
+        str(mycity_request)
+    )
+    mycity_response = MyCityResponseDataModel()
+    
+    r = requests.get("https://api2.community-boating.org/api/flag")
+    
+    mycity_response.output_speech = get_sailing_conditions(r.text)
+    mycity_response.session_attributes = mycity_request.session_attributes
+    mycity_response.card_title = mycity_request.intent_name
```
Comment (jmartini) [inline mycity/mycity/intents/sailing_weather_intent.py]: Let's make this a little more user friendly. Maybes something like "Community Boating Sailing Conditions"

File: mycity/mycity/intents/sailing_weather_intent.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+def get_sailing_conditions(api_response):
+    """
+    Checks for validity and parses the string returned from the API
+    
+    :param api_response: string indicating current sailing conditions
+    :return: string with message detailing current sailing conditions
+    """
+    server_flag_msg = api_response  # Ex:     var FLAG_COLOR = "Y"
+    if isinstance(server_flag_msg,str) and len(server_flag_msg) >= 19:
+        trimmed_flag_msg = server_flag_msg[18]  # Will produce a letter: G,Y,R,C
```
Comment (jmartini) [inline mycity/mycity/intents/sailing_weather_intent.py]: I think we can make the intention here more clear then just indexing into the string. How about using Python's string split or even a regular expression?

File: mycity/mycity/intents/sailing_weather_intent.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+def get_sailing_conditions(api_response):
+    """
+    Checks for validity and parses the string returned from the API
+    
+    :param api_response: string indicating current sailing conditions
+    :return: string with message detailing current sailing conditions
+    """
+    server_flag_msg = api_response  # Ex:     var FLAG_COLOR = "Y"
+    if isinstance(server_flag_msg,str) and len(server_flag_msg) >= 19:
```
Comment (jmartini) [inline mycity/mycity/intents/sailing_weather_intent.py]: I would make this slightly more future proof and not depend on string length. What about something like `"FLAG_COLOR" in server_flag_msg`

File: mycity/mycity/intents/sailing_weather_intent.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                                             + " Charles River.")
+        elif trimmed_flag_msg == 'R':
+            output = ("The sailing conditions are currently"
+                                             + " red at Community Boating on the"
+                                             + " Charles River.")
+        elif trimmed_flag_msg == 'C':
+            output = ("Community Boating on the Charles"
+                                             + " River is currently closed.")
+        else:
+            output = ("Something went wrong.")
```
Comment (jmartini) [inline mycity/mycity/intents/sailing_weather_intent.py]: I think we should make this a bit more conversational as suggested by Amazon's [interaction guidelines](https://developer.amazon.com/designing-for-voice/what-alexa-says/). Something like "I can't get that information right now. Try again later"

File: mycity/mycity/intents/sailing_weather_intent.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+            output = ("The sailing conditions are currently"
+                                             + " yellow at Community Boating on the"
+                                             + " Charles River.")
+        elif trimmed_flag_msg == 'R':
+            output = ("The sailing conditions are currently"
+                                             + " red at Community Boating on the"
+                                             + " Charles River.")
+        elif trimmed_flag_msg == 'C':
+            output = ("Community Boating on the Charles"
+                                             + " River is currently closed.")
```
Comment (jmartini) [inline mycity/mycity/intents/sailing_weather_intent.py]: Not required, but pulling these scripts out of the code and into a constants/data file or even as constants at the top of this page will make it easier to change the script, as well as use it in any testing code.

File: mycity/mycity/intents/sailing_weather_intent.py (hunk 1 above)
Comment (GreenGiraffe1) [reply mycity/mycity/intents/sailing_weather_intent.py]: Hey Jamie, are you referring to the name of the intent, or the name of the method here?

Comment (jmartini) [reply mycity/mycity/intents/sailing_weather_intent.py]: Whatever we set `mycity_response.card_title`. This is what is displayed on the users device if their Alexa has a screen

**Comment by GreenGiraffe1** [comment]
Closed, not relevant enough to project

**Pull request closed without merging**